src/app/modules/main/chat_section/chat_content/controller.nim

```diff
@@ -184,12 +184,6 @@ proc init*(self: Controller) =
       return
     self.delegate.onGroupChatDetailsUpdated(args.newName, args.newColor, args.newImage)
 
-  self.events.on(SIGNAL_CHAT_UPDATE) do(e: Args):
-    var args = ChatUpdateArgs(e)
-    for chat in args.chats:
-      if self.chatId == chat.id:
-        self.delegate.onChatEdited(chat)
```

Comment:
why this signal doesn't need to be handled here?

Reply:
I made it so that it's the parent (`chat_section`) that calls `onChatUpdated` instead. That way, the `chat_section` module can pass us the right chatItem. t reduces the duplication of code.